source/game.figure.bmx

```diff
@@ -539,6 +528,7 @@ Type TFigure extends TFigureBase
 	'override to add support for rooms
 	Method IsInRoom:Int(roomName:String="")
 		If roomName
+			Local _inRoom:TRoomBase = inRoom
```

Comment:
Variable _inRoom dann nicht verwendet

Reply:
Da war der Plan wohl, auf die einmal eingelesene Variable zuzugreifen, im Fall von paralleler Aenderung von "inRoom" (ja, das lesen waere hier auch nicht threadsafe ... sowas geht nur ueber "CAS" + einen "GetInRoom()"-Getter().

Reply:
Lustigerweise ist hier die nächste Exception geflogen - in Zeile 533 - also unmittelbar, nachdem schon geprüft wurde, ob room gesetzt ist.

Reply:
Ich habe den Code hier mal vervollstaendigt ... Gedanke ist: einmalig die Variable zu holen (wenn die gerade ueberschrieben wuerde..waere sie halt invalid/null ..)